### PR TITLE
fix n_iter reset

### PR DESCRIPTION
**Pull request by Gabriel-SGama**
The variable n_iter wasn't being reset every epoch, so the mean training loss was lower than it should be.

I removed the previous pull request because I wasn't sure if this would be the proper manner to correct this mistake since it was just one line (opening an issue would end in the same way).

Also, thank you for sharing the code. It helps to replicate results and unify evaluation.

### CODE REVIEW

**Comment by alessandrodepalma** [comment]
As I commented on your earlier pull request, many thanks for spotting this. 

Indeed, this influenced some of our secondary plots in the paper: we updated the manuscript and added you to the acknowledgments. The updated version should be up on arXiv in a couple of days.
